### ISSUE TITLE
Train_model.py

### ISSUE THREAD

**Issue by shweta-cmyk** [shweta-cmyk/Real-Time-Face-Mask-Detection-Using-CNN#3]
import os
os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2'

import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Conv2D, MaxPooling2D, Flatten, Dense, Dropout, Input
from tensorflow.keras.preprocessing.image import ImageDataGenerator

# ===============================
# Dataset paths
# ===============================
TRAIN_DIR = "Face Mask Dataset/train"
VAL_DIR = "Face Mask Dataset/validation"

IMG_SIZE = 128
BATCH_SIZE = 32
EPOCHS = 5

# ===============================
# Data generators
# ===============================
train_datagen = ImageDataGenerator(
    rescale=1./255,
    rotation_range=20,
    zoom_range=0.2,
    shear_range=0.2,
    horizontal_flip=True
)

val_datagen = ImageDataGenerator(rescale=1./255)

train_generator = train_datagen.flow_from_directory(
    TRAIN_DIR,
    target_size=(IMG_SIZE, IMG_SIZE),
    batch_size=BATCH_SIZE,
    class_mode='binary'
)

val_generator = val_datagen.flow_from_directory(
    VAL_DIR,
    target_size=(IMG_SIZE, IMG_SIZE),
    batch_size=BATCH_SIZE,
    class_mode='binary'
)

# ===============================
# CNN Model (FIXED)
# ===============================
model = Sequential([
    Input(shape=(IMG_SIZE, IMG_SIZE, 3)),

    Conv2D(32, (3, 3), activation='relu'),
    MaxPooling2D(2, 2),

    Conv2D(64, (3, 3), activation='relu'),
    MaxPooling2D(2, 2),

    Conv2D(128, (3, 3), activation='relu'),
    MaxPooling2D(2, 2),

    Flatten(),
    Dense(128, activation='relu'),
    Dropout(0.5),
    Dense(1, activation='sigmoid')
])

# ===============================
# Compile
# ===============================
model.compile(
    optimizer='adam',
    loss='binary_crossentropy',
    metrics=['accuracy']
)

# ===============================
# Train
# ===============================
model.fit(
    train_generator,
    epochs=EPOCHS,
    validation_data=val_generator
)

# ===============================
# Save model (H5)
# ===============================
model.save("face_mask_detector_model.h5")

print("✅ Training completed")
print("✅ Model saved as face_mask_detector_model.h5")
